### PR TITLE
[MRG] Forward argument axes from plot_sensors to DigMontage.plot

### PR DESCRIPTION
**Pull request by mscheltienne**
This PR enables:

```
montage = ...  # DigMontage
f, ax = plt.subplots(1, 1)
montage.plot(axes=ax)
```

I noticed it was missing when I tried to plot 3 montages on the same figure. One more step towards a unified plotting API :smiley: 

Closes #11404

### CODE REVIEW

**Comment by mscheltienne** [comment]
Rendered documentation:
- [DigMontage.plot](https://output.circle-artifacts.com/output/job/25fab575-b69e-4a72-b194-0483b3541e82/artifacts/0/dev/generated/mne.channels.DigMontage.html#mne.channels.DigMontage.plot)
- [plot_sensors](https://output.circle-artifacts.com/output/job/25fab575-b69e-4a72-b194-0483b3541e82/artifacts/0/dev/generated/mne.viz.plot_sensors.html#mne-viz-plot-sensors)

File: mne/channels/tests/test_montage.py [hunk 1]
```diff
@@ -1644,6 +1644,10 @@ def test_plot_montage():
     montage.plot()
     plt.close('all')
 
+    f, ax = plt.subplots(1, 1)
+    montage.plot(axes=ax)
```
Comment (larsoner) [inline mne/channels/tests/test_montage.py:1648]: Can you add 2-3 `with pytest.raises` here to hit the exceptions you added, too?

**Comment by mscheltienne** [comment]
@larsoner Agree for the comments, it's just that we need to be careful since Axes3D is an instance of Axes. But should I address those now.. I was a bit hasty on this, I now notice an on-going PR about this issue: #11404
We can wait for #11404 to be merged first, and I can add the additional changes on the updated main.

**Comment by larsoner** [comment]
That one was opened a month ago and never returned to. You could merge their changes into yours, or start (over) with their changes/branch and put yours on top, then we can merge giving you both credit.

**Comment by mscheltienne** [comment]
Failures related to https://github.com/mne-tools/mne-python/pull/11454 maybe?

**Comment by drammock** [comment]
> Failures related to #11454 maybe?

I opened #11476 to track these failures.

**Comment by mscheltienne** [comment]
It should be good to merge. Thanks for the fix in #11478!

**Comment by drammock** [comment]
`axes` param is documented twice
https://output.circle-artifacts.com/output/job/b44bcf3f-92ce-4799-95b2-7979266997d8/artifacts/0/dev/generated/mne.viz.plot_montage.html#mne-viz-plot-montage

TBH I would have expected that to raise an error :/

File: mne/viz/montage.py [hunk 2]
```diff
@@ -25,7 +25,15 @@ def plot_montage(montage, scale_factor=20, show_names=True, kind='topomap',
...
     %(sphere_topomap_auto)s
+    %(axes_montage)s
+
+        .. versionadded:: 1.4
     %(verbose)s
+    axes : instance of Axes | instance of Axes3D | None
+        Axes to draw the sensors to. If ``kind='3d'``, axes must be an instance
+        of Axes3D. If None (default), a new axes will be created.
+
+        .. versionadded:: 1.4.0
```
Comment (larsoner) [inline mne/viz/montage.py:36]:
```suggestion
```

**Comment by mscheltienne** [comment]
Arf.. that's from the merge of #11404 🙈 Thanks @drammock !

**Pull request merged**